lib/demo/interactors/create_comment.rb

```diff
@@ -0,0 +1,23 @@
+require 'lotus/interactor'
+
+module Demo
+  class CreateComment
+    include Lotus::Interactor
+
+    def initialize(params:, repository: CommentRepository)
+      @repository = repository
+      @params = params
+    end
+
+    def call
```

Comment:
The reason behind this design choice is to have `#call` as a descriptor of the needed actions to take to complete this interaction and to let its implementation to act on the same level of abstraction.

This is a pattern called [Template Method](http://en.wikipedia.org/wiki/Template_method_pattern), where a routine describes the steps to take to finish an algorithm. The implementation of these steps is delegated to private routines or other classes.

The other reason is that, at a first glance you can understand that this use case does. Our example is simple, but think if we need to send email notifications, expire caches, update stats, push with external analytics services. That method will become soon a mess, but with this approach you can have:

```ruby
def call
  persist_comment
  deliver_email_notifications
  sync_with_intercom
end
```